plugins/org.komodo.shell/src/org/komodo/shell/commands/core/ImportCommand.java

```diff
@@ -50,33 +61,85 @@ public boolean execute() throws Exception {
-            String modelNameArg = requiredArgument(2, Messages.getString("ImportCommand.InvalidArgMsg_ModelName")); //$NON-NLS-1$
+            String parentContextPath = optionalArgument(2);
 
-            ImportMessages messages = importModel(fileNameArg, modelNameArg);
-            if (!messages.hasError()) {
-                print(CompletionConstants.MESSAGE_INDENT, Messages.getString("ImportCommand.ModelImportSuccessMsg", modelNameArg, fileNameArg)); //$NON-NLS-1$
-                if (getWorkspaceStatus().getRecordingStatus())
-                    recordCommand(getArguments());
-            } else {
-                print(CompletionConstants.MESSAGE_INDENT, Messages.getString("ImportCommand.ImportFailedMsg", fileNameArg)); //$NON-NLS-1$
-                print(CompletionConstants.MESSAGE_INDENT, messages.errorMessagesToString());
+            // Validate fileName
+            if(!validateFileName(fileNameArg)) {
+            	return false;
+            }
+            // Validate VDB is valid for the specified parent
+            if(!validateParentPath(parentContextPath,KomodoType.VDB)) {
+            	return false;
             }
-        } else if (SUBCMD_VDB.equalsIgnoreCase(subcmdArg)) {
-            // Check required args
-            String fileNameArg = requiredArgument(1, Messages.getString("ImportCommand.InvalidArgMsg_FileName")); //$NON-NLS-1$
 
-            ImportMessages messages = importVdb(fileNameArg);
-            if (!messages.hasError()) {
-                print(CompletionConstants.MESSAGE_INDENT, Messages.getString("ImportCommand.VdbImportSuccessMsg", fileNameArg)); //$NON-NLS-1$
-                if (getWorkspaceStatus().getRecordingStatus())
-                    recordCommand(getArguments());
+            KomodoObject vdbObject = importVdb(fileNameArg,parentContextPath);
+            
+            if(vdbObject!=null && !importMessages.hasError()) {
+            	Vdb theVdb = (Vdb)vdbObject;
+            	String vdbName = theVdb.getVdbName(wsStatus.getTransaction());
+            	
+            	// Check for already existing vdb with same name
+            	WorkspaceContext parentContext = ContextUtils.getContextForPath(wsStatus, parentContextPath);
+            	boolean okToImport = validateNotDuplicate(vdbName, KomodoType.VDB, parentContext);
+            	// OK to keep VDB - rename it to vdbName
+            	if(okToImport) {
+            		vdbObject.rename(wsStatus.getTransaction(), vdbName);
+            	// Has Duplicate VDB - delete it.
+            	} else {
+                    WorkspaceManager wkspManager = wsStatus.getCurrentContext().getWorkspaceManager();
+                    wkspManager.delete(getWorkspaceStatus().getTransaction(), vdbObject);
+            	}
+                               
+                // Commit transaction
+                getWorkspaceStatus().commit("ImportCommand"); //$NON-NLS-1$
```

Comment:
Seems like we should only commit if we want to persist the import and the subsequent rename. If we found a duplicate, we should rollback the transaction instead of doing a delete and commit.